modules/apps/transfer/keeper/msg_server.go

```diff
@@ -58,17 +58,14 @@ func (k Keeper) Transfer(goCtx context.Context, msg *types.MsgTransfer) (*types.
 	}
 
 	events := make([]sdk.Event, 0, len(tokens)+1)
-	for _, token := range tokens {
-		k.Logger(ctx).Info("IBC fungible token transfer", "token", token.Denom, "amount", token.Amount.String(), "sender", msg.Sender, "receiver", msg.Receiver)
-		events = append(events, sdk.NewEvent(
-			types.EventTypeTransfer,
-			sdk.NewAttribute(types.AttributeKeySender, msg.Sender),
-			sdk.NewAttribute(types.AttributeKeyReceiver, msg.Receiver),
-			sdk.NewAttribute(types.AttributeKeyDenom, token.Denom),
-			sdk.NewAttribute(types.AttributeKeyAmount, token.Amount.String()),
-			sdk.NewAttribute(types.AttributeKeyMemo, msg.Memo),
-		))
-	}
+	k.Logger(ctx).Info("IBC fungible token transfer", "tokens", tokens, "sender", msg.Sender, "receiver", msg.Receiver)
+	events = append(events, sdk.NewEvent(
+		types.EventTypeTransfer,
+		sdk.NewAttribute(types.AttributeKeySender, msg.Sender),
+		sdk.NewAttribute(types.AttributeKeyReceiver, msg.Receiver),
+		sdk.NewAttribute(types.AttributeKeyTokens, tokens.String()),
+		sdk.NewAttribute(types.AttributeKeyMemo, msg.Memo),
+	))
```

Comment:
should we still be creating a slice with underlying array `len(tokens)+1` and appending?  we're no longer ranging over `tokens`

I think the `[]sdk.Event` should actually be 2 now. One for the transfer event and one for the usual `EventTypeMessage` with module key/val.

Reply:
yes indeed. I'm just going to remove the `events` variable altogether and do direct emission

Reply:
done https://github.com/cosmos/ibc-go/pull/6348/commits/4ed0cb007616f73d893eb90ee001ccd7cb5d6f05